domain/src/test/java/org/oppia/domain/exploration/ExplorationDataControllerTest.kt

```diff
@@ -0,0 +1,223 @@
+
+  // https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/
+  @ObsoleteCoroutinesApi
+  private val testThread = newSingleThreadContext("TestMain")
+
+  @Before
+  @ExperimentalCoroutinesApi
+  @ObsoleteCoroutinesApi
+  fun setUp() {
+    Dispatchers.setMain(testThread)
+    setUpTestApplicationComponent()
+  }
+
+  @After
+  @ExperimentalCoroutinesApi
+  @ObsoleteCoroutinesApi
+  fun tearDown() {
+    Dispatchers.resetMain()
+    testThread.close()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerExplorationDataControllerTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testController_providesInitialLiveDataFortheWelcomeExploration() = runBlockingTest(coroutineContext) {
+    val explorationLiveData = explorationDataController.getExplorationById(TEST_EXPLORATION_ID_0)
+    advanceUntilIdle()
+    assertThat(explorationLiveData).isNotNull()
+    explorationLiveData!!.observeForever(mockExplorationObserver)
+    val expectedExplorationStateSet = listOf("END", "Estimate 100", "Numeric input",
+      "Things you can do", "Welcome!", "What language")
+
+    verify(mockExplorationObserver, atLeastOnce()).onChanged(explorationResultCaptor.capture())
+    assertThat(explorationResultCaptor.value.isSuccess()).isTrue()
+    assertThat(explorationResultCaptor.value.getOrThrow()).isNotNull()
+    val exploration = explorationResultCaptor.value.getOrThrow();
+    assertThat(exploration.title).isEqualTo("Welcome to Oppia!")
+    assertThat(exploration.languageCode).isEqualTo("en")
+    assertThat(exploration.statesCount).isEqualTo(6)
+    assertThat(exploration.statesMap.keys).containsAllIn(expectedExplorationStateSet)
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testController_providesInitialLiveDataFortheAboutOppiaExploration()
+      = runBlockingTest(coroutineContext) {
+    val explorationLiveData = explorationDataController.getExplorationById(TEST_EXPLORATION_ID_1)
+    advanceUntilIdle()
+    assertThat(explorationLiveData).isNotNull()
+    explorationLiveData!!.observeForever(mockExplorationObserver)
+    val expectedExplorationStateSet = listOf("About this website", "Contact", "Contribute", "Credits", "END",
+      "End Card", "Example1", "Example3", "First State", "Site License", "So what can I tell you")
+
+    verify(mockExplorationObserver, atLeastOnce()).onChanged(explorationResultCaptor.capture())
+    assertThat(explorationResultCaptor.value.isSuccess()).isTrue()
+    assertThat(explorationResultCaptor.value.getOrThrow()).isNotNull()
+    val exploration = explorationResultCaptor.value.getOrThrow();
+    assertThat(exploration.title).isEqualTo("About Oppia")
+    assertThat(exploration.languageCode).isEqualTo("en")
+    assertThat(exploration.statesCount).isEqualTo(11)
+    assertThat(exploration.statesMap.keys).containsAllIn(expectedExplorationStateSet)
```

Comment:
Nit: you could alternatively use ``containsExactlyIn`` which implies the count, as well.

Reply:
Done